pkg/collector/corechecks/snmp/internal/devicecheck/devicecheck.go

```diff
@@ -55,6 +56,60 @@ const (
 	checkDurationThreshold  = 30 // Thirty seconds
 )
 
+type profileCache struct {
+	sysObjectID string
+	timestamp   time.Time
+	profile     *profiledefinition.ProfileDefinition
+	err         error
+	scalarOIDs  []string
+	columnOIDs  []string
+}
+
+// GetProfile returns the cached profile, or an empty profile if the cache is empty.
+// Use this when you need to make sure you have *some* profile.
+func (pc *profileCache) GetProfile() profiledefinition.ProfileDefinition {
+	if pc.profile == nil {
+		return profiledefinition.ProfileDefinition{
+			Metadata: make(profiledefinition.MetadataConfig),
+		}
+	}
+	return *pc.profile
+}
+
+func (pc *profileCache) Update(sysObjectID string, now time.Time, config *checkconfig.CheckConfig) (profiledefinition.ProfileDefinition, error) {
+	if pc.IsOutdated(sysObjectID, config.ProfileName, config.ProfileProvider.LastUpdated()) {
+		// we cache the value even if there's an error, because an error indicates that
+		// the ProfileProvider couldn't find a match for either config.ProfileName or
+		// the given sysObjectID, and we're going to have the same error if we call this
+		// again without either the sysObjectID or the ProfileProvider changing.
```

Comment:
we can also have an error when the `ProfileProvider` doesn't know `ProfileName` which can happen in the case the profile was not "loaded" yet